core/lib/dal/src/storage_logs_dedup_dal.rs

```diff
@@ -133,4 +133,27 @@ impl StorageLogsDedupDal<'_, '_> {
         .map(|row| H256::from_slice(&row.hashed_key))
         .collect()
     }
+
+    pub async fn enum_indices_for_keys(&mut self, hashed_keys: &[H256]) -> Vec<u64> {
+        let hashed_keys_order: HashMap<_, _> = hashed_keys
+            .iter()
+            .enumerate()
+            .map(|(index, key)| (key, index))
+            .collect();
+        let hashed_keys: Vec<_> = hashed_keys.iter().map(H256::as_bytes).collect();
+        let mut indices: Vec<(H256, u64)> = sqlx::query!(
+            "SELECT hashed_key, index FROM initial_writes \
+            WHERE hashed_key = ANY($1)",
+            &hashed_keys as &[&[u8]]
+        )
```

Comment:
```suggestion
        )
        .instrument("enum_indices_for_keys")
        .report_latency()
```

Would probably be good to instrument - not 100% sure this will compile (didn't test) - please have a look. We can let the CI figure it out as well :)